dynamic_programing/Knapsack.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @function knapsack
+ * @description Given weights and values of n (numberOfItems) items, put these items in a knapsack of capacity totalWeight to get the maximum total value in the knapsack. In other words, given two integer arrays values[0..n-1] and weights[0..n-1] which represent values and weights associated with n items respectively. Also given an integer totalWeight which represents knapsack capacity, find out the maximum value subset of values[] such that sum of the weights of this subset is smaller than or equal to totalWeight. You cannot break an item, either pick the complete item or don’t pick it (0-1 property).
+ * @Complexity_Analysis
+ * Space complexity - O(1)
```

Comment:
The space complexity is the same as the time complexity- you're building a full lookup table. It certainly isn't constant space complexity.

Reply:
yeah sorry for that, it's a big mistake. I am correcting that.